xbmc/guilib/GUIColorButtonControl.cpp

```diff
@@ -0,0 +1,168 @@
+  if (IsDisabled())
+    m_imgColorDisabledMask->Render();
+  else
+    m_imgColorMask->Render();
+}
+
+void CGUIColorButtonControl::Process(unsigned int currentTime, CDirtyRegionList& dirtyregions)
+{
+  m_imgColorMask->Process(currentTime);
+  m_imgColorDisabledMask->Process(currentTime);
+  CGUIButtonControl::Process(currentTime, dirtyregions);
+}
+
+bool CGUIColorButtonControl::OnAction(const CAction& action)
+{
+  return CGUIButtonControl::OnAction(action);
+}
+
+bool CGUIColorButtonControl::OnMessage(CGUIMessage& message)
+{
+  return CGUIButtonControl::OnMessage(message);
+}
+
+void CGUIColorButtonControl::AllocResources()
+{
+  CGUIButtonControl::AllocResources();
+  m_imgColorMask->AllocResources();
+  m_imgColorDisabledMask->AllocResources();
+  SetPosition(m_posX, m_posY);
+}
+
+void CGUIColorButtonControl::FreeResources(bool immediately)
+{
+  CGUIButtonControl::FreeResources(immediately);
+  m_imgColorMask->FreeResources(immediately);
+  m_imgColorDisabledMask->FreeResources(immediately);
+}
+
+void CGUIColorButtonControl::DynamicResourceAlloc(bool bOnOff)
+{
+  CGUIControl::DynamicResourceAlloc(bOnOff);
+  m_imgColorMask->DynamicResourceAlloc(bOnOff);
+  m_imgColorDisabledMask->DynamicResourceAlloc(bOnOff);
+}
+
+void CGUIColorButtonControl::SetInvalid()
+{
+  CGUIButtonControl::SetInvalid();
+  m_imgColorMask->SetInvalid();
+  m_imgColorDisabledMask->SetInvalid();
+}
+
+void CGUIColorButtonControl::SetPosition(float posX, float posY)
+{
+  CGUIButtonControl::SetPosition(posX, posY);
+  float colorPosX =
+      m_colorPosX ? m_posX + m_colorPosX : (m_posX + m_width - 8) - m_imgColorMask->GetWidth();
+  float colorPosY =
+      m_colorPosY ? m_posY + m_colorPosY : m_posY + (m_height - m_imgColorMask->GetHeight()) / 2;
+  m_imgColorMask->SetPosition(colorPosX, colorPosY);
+  m_imgColorDisabledMask->SetPosition(colorPosX, colorPosY);
+}
+
+void CGUIColorButtonControl::SetColorDimensions(float posX, float posY, float width, float height)
+{
+  m_colorPosX = posX;
+  m_colorPosY = posY;
+  if (width)
+  {
+    m_imgColorMask->SetWidth(width);
+    m_imgColorDisabledMask->SetWidth(width);
+  }
+  if (height)
+  {
+    m_imgColorMask->SetHeight(height);
+    m_imgColorDisabledMask->SetHeight(height);
+  }
+  SetPosition(GetXPosition(), GetYPosition());
+}
+
+void CGUIColorButtonControl::SetWidth(float width)
+{
+  CGUIButtonControl::SetWidth(width);
+  SetPosition(GetXPosition(), GetYPosition());
+}
+
+void CGUIColorButtonControl::SetHeight(float height)
+{
+  CGUIButtonControl::SetHeight(height);
+  SetPosition(GetXPosition(), GetYPosition());
+}
+
+std::string CGUIColorButtonControl::GetDescription() const
+{
+  return CGUIButtonControl::GetDescription();
+}
+
+void CGUIColorButtonControl::SetImageBoxColor(KODI::GUILIB::GUIINFO::CGUIInfoColor color)
+{
+  m_imgBoxColor = color;
+}
+
+void CGUIColorButtonControl::SetImageBoxColor(std::string hexColor)
+{
+  if (hexColor.empty())
+    m_imgBoxColor = UTILS::COLOR::BLACK;
```

Comment:
Out of curiosity, any reason to fill it black by default if no color is defined? I'd prefer it not to have any color (transparency) if the color is not explicitly set